### ISSUE TITLE
Extend exception message of checks of classifier/attack compatibility to provide better/more detailed explanation

### ISSUE THREAD

**Issue by HaqueNurImtiazul** [IBM/adversarial-robustness-toolbox#174]
```
import pandas as pd
from sklearn.svm import SVC
import numpy as np
from sklearn.model_selection import train_test_split
from art.attacks import FastGradientMethod
from art.classifiers import SklearnClassifier
from art.classifiers import scikitlearn
from art.utils import load_mnist
from sklearn import tree
from sklearn.ensemble import RandomForestClassifier
from sklearn.neural_network import MLPClassifier
from sklearn.tree import DecisionTreeClassifier


# Importing the dataset
dataset = pd.read_csv('dataset.csv')

inputData = dataset.iloc[:, :-1].values
outputData = dataset.iloc[:, 17].values

# Splitting the dataset into the Training set and Test set

x_train, x_test, y_train, y_test = train_test_split(inputData, outputData, test_size = .25, random_state = 0)

# Creating model

model = RandomForestClassifier(n_estimators=500, random_state=0)

# Step 3: Create the ART classifier

classifier = scikitlearn.ScikitlearnRandomForestClassifier(model=model)

# Step 4: Train the ART classifier

classifier.fit(x_train, y_train)

# Step 5: Evaluate the ART classifier on benign test examples

predictions = classifier.predict(x_test)
accuracy = np.sum(np.argmax(predictions, axis=1) == (y_test)) / len(y_test)
print('Accuracy on benign test examples: {}%'.format(accuracy * 100))

# Step 6: Generate adversarial test examples
attack = FastGradientMethod(classifier=classifier, eps=0.2)
x_test_adv = attack.generate(x=x_test)

# Step 7: Evaluate the ART classifier on adversarial test examples

predictions = classifier.predict(x_test_adv)
accuracy = np.sum(np.argmax(predictions, axis=1) == y_test) / len(y_test)
print('Accuracy on adversarial test examples: {}%'.format(accuracy * 100))
```

**Error message:**
For `FastGradientMethod` classifier must be an instance of `art.classifiers.classifier.ClassifierGradients`, the provided classifier is instance of (<class 'art.classifiers.scikitlearn.ScikitlearnClassifier'>,).

(Dataset is attached)
[dataset.txt](https://github.com/IBM/adversarial-robustness-toolbox/files/3665147/dataset.txt)

**Comment by beat-buesser** [comment]
Hi @HaqueNurImtiazul Thank you for using ART and please apologise the delayed response. ART is raising an exception, leading to the error in the script above, because you are trying to use an attack (`FastGradientMethod`) which requires gradients with respect to the input and you are providing a classifier (`ScikitlearnRandomForestClassifier`) which does not provide gradients with respect to the inputs, which means `FastGradientMethod` and `ScikitlearnRandomForestClassifier` cannot be used together. If you would like to use `ScikitlearnRandomForestClassifier` you could consider a black-box attack like `ZooAttack`, which does not require the classifier to provide gradients.

However, I think you might have a good point here and we should rename this issue to change the error message so that it explains the cause of this exception better and in more detail for future users.

**Comment by HaqueNurImtiazul** [comment]
Which classifier can be used with FastGradientMethod?

**Comment by beat-buesser** [comment]
All classifier's providing gradients through API `ClassifierGradients`, e.g. `KerasClassifier`, `MXClassifier`, `PyTorchClassifier`, `TensorFlowClassifier`, `TensorFlowV2Classifier`, `ScikitlearnLogisticRegression`, `ScikitlearnSVC`
